linearElasticity/BoussinesqProblem/BoussinesqProblemAnalyticalSolution/Make/options

```diff
@@ -0,0 +1,24 @@
+ifeq ($(WM_PROJECT), foam)
+    VERSION_SPECIFIC_INC = -DFOAMEXTEND
+else
+    VERSION_SPECIFIC_INC = -DOPENFOAMESIORFOUNDATION
+    ifneq (,$(findstring v,$(WM_PROJECT_VERSION)))
+        VERSION_SPECIFIC_INC += -DOPENFOAMESI
+    else
+        VERSION_SPECIFIC_INC += -DOPENFOAMFOUNDATION
+    endif
+endif
+
+SOLIDS4FOAM_MODELS=../../../../solids4foam-ib/src/solids4FoamModels/lnInclude
```

Comment:
How can we above hardcoding this?

We know the relative address in the main solids4foam repo, but here we don't. One suggestion is to insist that the user sets the SOLIDS4FOAM_DIR variable (we can check this in the Allwmake script) and then use that.

Reply:
I did not find a proper solution for this. Is it possible to set SOLIDS4FOAM_DIR absolute path variable automatically when invoking Allwmake?

Reply:
By the way, I was getting an error when running the sample command: "compound has already been transfered from token", just like at https://www.cfd-online.com/Forums/openfoam-cc-toolkits-fluid-structure-interaction/238595-compound-has-already-been-transfered-token.html.

The problem does not seem to be in analyticalBoussinesqProblemDisplacement but instead in fixedDisplacement, where "value" is read twice (which seems to cause this issue). I push a fix to this in the solids4foam development branch: https://github.com/solids4foam/solids4foam/commit/6f71ef560d1bbff121eaa58016638b7ef588070b

Reply:
Nice solution, thanks! 

I never got this token error, glad that you solved also that!